demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -1,1081 +1,557 @@
+                      uint32_t ulCommandResponsePayloadBufferSize)
+{
+    AzureIoTResult_t xResult;
+    AzureIoTJSONReader_t xReader;
+    AzureIoTJSONWriter_t xWriter;
+    int32_t lCommandNameLength;
+    int32_t ulCommandResponsePayloadLength;
+
+    LogInfo( ( "Command payload : %.*s \r\n",
+               pxMessage->ulPayloadLength,
+               ( const char * ) pxMessage->pvMessagePayload ) );
+
+    lCommandNameLength = sizeof( sampleazureiotCOMMAND_MAX_MIN_REPORT ) - 1;
+
+    if( ( lCommandNameLength == pxMessage->usCommandNameLength ) &&
+        ( strncmp( sampleazureiotCOMMAND_MAX_MIN_REPORT, ( const char * ) pxMessage->pucCommandName, lCommandNameLength ) == 0 ) )
+    {
+        /* Is for max min report */
+
+        /*Initialize the reader from which we pull the "since". */
+        xResult = AzureIoTJSONReader_Init( &xReader, pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+        configASSERT( xResult == eAzureIoTSuccess );
+
+        /* Initialize the JSON writer with a buffer to which we will write the response payload. */
+        xResult = AzureIoTJSONWriter_Init( &xWriter, ucCommandResponsePayloadBuffer, ulCommandResponsePayloadBufferSize );
+        configASSERT( xResult == eAzureIoTSuccess );
+
+        /* Read from the writer the "since" value and use it to construct the response payload in the writer. */
+        xResult = prvInvokeMaxMinCommand( &xReader, &xWriter );
+
+        if( xResult == eAzureIoTSuccess )
+        {
+            ulCommandResponsePayloadLength = AzureIoTJSONWriter_GetBytesUsed( &xWriter );
+
+            *ulResponseStatus = 200;
+        }
+        else
+        {
+            LogError( ( "Error generating command payload: result 0x%08x", xResult ) );
+
+            *ulResponseStatus = 501;
+            ulCommandResponsePayloadLength = sizeof( sampleazureiotCOMMAND_EMPTY_PAYLOAD ) - 1;
+            (void)memcpy( ucCommandResponsePayloadBuffer, sampleazureiotCOMMAND_EMPTY_PAYLOAD, ulCommandResponsePayloadLength );
+        }
+    }
+    else
+    {
+        /* Not for max min report (not for this device) */
+        LogInfo( ( "Received command is not for this device: %.*s",
+                   pxMessage->usCommandNameLength,
+                   pxMessage->pucCommandName ) );
+
+        *ulResponseStatus = 404;
+        ulCommandResponsePayloadLength = sizeof( sampleazureiotCOMMAND_EMPTY_PAYLOAD ) - 1;
+        (void)memcpy( ucCommandResponsePayloadBuffer, sampleazureiotCOMMAND_EMPTY_PAYLOAD, ulCommandResponsePayloadLength );
+    }
+
+    return ulCommandResponsePayloadLength;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Implements the sample interface for generating Telemetry payload.
+ */
+uint32_t ulCreateTelemetry( uint8_t * pucTelemetryData,
+                            uint32_t ulTelemetryDataLength )
+{
+    return snprintf( ( char * ) pucTelemetryData, ulTelemetryDataLength,

```

Comment:
>pucTelemetryData [](http://example.com/codeflow?start=32&length=16)

return value of snprintf is:

1. Number of bytes written into buffer, when buffer space is enough.
2. Number of bytes writtern into bufffer, when buffer space is not enough. Only difference from case 1 is : number of bytes in this case is trucated bytes written.
3. Negative value for other errors.


This is little different from what we expects in this function. Main reason is,  because we merge case 1 and case 2 return values. Can we check if the data is writen fully, only then return success otherwise return failure. Also, return value should be int32_t to support negative value as error?